packages/core/admin/admin/src/pages/Settings/pages/Webhooks/tests/ListPage.test.tsx

```diff
@@ -17,6 +18,18 @@ jest.mock('@strapi/helper-plugin', () => ({
   useFocusWhenNavigate: jest.fn(),
 }));
 
+jest.mock('react-router-dom', () => ({
+  ...jest.requireActual('react-router-dom'), 
+  useHistory: () => ({
+    push: mockedPush
+  }),
+  useLocation: jest.fn().mockReturnValue({
+    pathname: '/admin/settings/webhooks',
+  })
+}));
+const mockedPush = jest.fn();
```

Comment:
I would use a pattern like this instead of mocking: https://github.com/strapi/strapi/blob/1cbd1d176532464002c07c7bf6a3e39d0753faec/packages/core/admin/admin/src/features/tests/BackButton.test.tsx#L8 less mocking, and we'd need to make the change upstream in v5 regardless. But mainly, mocking is brittle.

Reply:
The problem is that if we don't mock pathname we get `//create` in the expected value, because `pathname = null`

Reply:
Nevermind 👍🏼